src/easymigration/scripts/list_bagstore_files.py

```diff
@@ -0,0 +1,84 @@
+# given a doi, retrieve the uuid and subsequently the files from the bag-store
+
+import argparse
+import csv
+import logging
+import sys
+
+import requests
+from xml.dom import minidom
+from easymigration.batch_processing import batch_process
+from easymigration.config import init
+from easymigration.pids_handling import non_empty_lines
+
+
+def find_files(doi, dark_archive, csv_writer):
+    # TODO continue batch if not found
+    uuid = find_uuid(doi, dark_archive["index_url"])
+    files_xml = get_files_xml(uuid, dark_archive["store_url"])
+    parse_files_xml(doi, files_xml, csv_writer)
+
+
+def find_uuid(doi, bag_index_url):
+    # locate in bag-index
+    try:
+        params = {"doi": doi}
+        response = requests.get(f"{bag_index_url}/search", params=params)
+        response.raise_for_status()
+    except requests.exceptions.RequestException as re:
+        logging.error("RequestException: ", re)
+        raise
+    resp_data = response.json()["result"][0]["bag-info"]["bag-id"]
+    return resp_data
+
+
+def get_files_xml(uuid, bag_store_url):
+    try:
+        url = f"{bag_store_url}/bags/{uuid}/metadata/files.xml"
+        logging.debug(url)
+        response = requests.get(url)
+        response.raise_for_status()
+    except requests.exceptions.RequestException as re:
+        logging.error("RequestException: ", re)
+        raise
+    return response.text
+
+
+def parse_files_xml(doi, files_xml, csv_writer):
+    file_items = minidom.parseString(files_xml).getElementsByTagName("file")
+    for elem in file_items:
+        access = elem.getElementsByTagName("accessibleToRights")
+        access_text = ""
+        if access is not None and len(access) == 1:
+            access_text = access[0].firstChild.nodeValue
+        row = {"doi": doi, "path": elem.attributes["filepath"].value, "accessCategory": access_text}
+        csv_writer.writerow(row)
+
+
+def main():
+    config = init()
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="For each doi, list the files in the bag-store"
+    )
+    parser.add_argument("-d", "--doi", dest="doi",
+                        help="Pid of a single dataset for which to find the files. "
+                             "When omitted, DOIs are read from stdin.")
+    args = parser.parse_args()
+
+    dark_archive = config["dark_archive"]
+
+    fieldnames = {"doi", "path", "accessCategory"}
+    csv_writer = csv.DictWriter(sys.stdout, delimiter=",", fieldnames=fieldnames)
+    csv_writer.writeheader()
+
+    if args.doi is not None:
+        find_files(args.doi, dark_archive, csv_writer)
+    else:
+        dois = non_empty_lines(sys.stdin.read())
```

Comment:
See https://dans-knaw.github.io/dans-datastation-tools/dev/#coding-style

Reply:
most scripts in dans-datastation-tools have these two parameters.  i don't really care which we use, as long as it is used consistently. @janvanmansum ?

Reply:
The issue also speaks about accepting both doi's as UUIDs. Should I recognize them from the input (for example by a `doi:` or `uuid:` prefix) or do you want different command line arguments? In the latter case: use `stdin` when the option has no value?

Reply:
@jo-pol : @lindareijnhoudt  and I since concluded that for some scripts you want to force the user to send the output to a file. I think I'll change the coding style policy. How about...

* Use options for input and output files.
* If the output is typically large or takes a long time to generate, make the output option a required option.
* `-` means stdin or stdout, e.g. `--output -` means write to the stdout anyway.

?

Reply:
A dash for std/stdout is just a convention which individual commands may or may not implement. I did now in `pid_handling.py` for easy reuse by future commands.